### PR TITLE
Cleanup Proposals

### PR DESCRIPTION
**Pull request by JonDouglas**
This PR cleans up the proposals from the original merge.

It includes fixed images, better explanations, and more.

It also includes a revamp to the proposal template to help describe the expectations of a proposal.

I think there's opportunity here to discuss what we'd like to see in a proposal template. I don't have a good answer, but gave it my best shot without making it too complex.

### CODE REVIEW

File: meta/template.md
Comment (loic-sharma) [inline meta/template.md]: This proposal process is open to third-parties, right? How will they determine ship dates? Even for folks on the NuGet team, this seems like a hard section to fill out as part of a proposal. What do you think of removing this section?

File: meta/template.md
Comment (loic-sharma) [inline meta/template.md:48]: What do you think of keeping the "References" section? In my mind, I want to keep the previous sections to the point so that the reader can follow my thought process. I wouldn't want to overwhelm the reader with many links in the "Problem: ..." or "Why: ..." sections. On the other hand, the "References" section at the bottom is the perfect place to drop miscellaneous links for folks that want to drill deeper on the topic.

File: meta/template.md
Comment (loic-sharma) [inline meta/template.md]:
```suggestion
A low-fidelity mockup or pseudocode that shows how this proposal would look.
```

File: meta/template.md
Comment (JonDouglas) [reply meta/template.md]: Great point. I think we should remove it then.

File: meta/template.md
Comment (JonDouglas) [reply meta/template.md:48]: I looked at other design repos & didn't find any references to be needed.

https://github.com/rust-lang/rfcs/tree/master/text
https://github.com/npm/rfcs/tree/latest/accepted
https://github.com/dotnet/designs/tree/main/accepted/2020

I think the bottom may need a Q&A section though?

Comment (JonDouglas) [reply meta/template.md:48]: Also, i just took a stab at the common things needed. Maybe there's a better combination from the various RFC & design processes already established.

Comment (loic-sharma) [reply meta/template.md:48]: Q&A would work too. FYI, I don't feel strongly about this feedback so feel free to skip it :)

Comment (nkolev92) [reply meta/template.md:48]: The `Considerations` and `References` sections were added as a response to a *need* in previous designs. 

In particular, the exact design is not the only thing that matters. Design that were *not* chosen, and the reasoning behind that is equally, if often times not more important. 

The `References` section is there to remind people that not everyone has the same context and that they should try to make the design as complete as possible. 

I think the `Considerations` section is especially valuable. I can see how `References` might not have as much value as others. You can always just add the references in the specific sub section they are necessary for.

File: meta/template.md
Comment (nkolev92) [inline meta/template.md]: Feels like the Description, Problem and Why sections are very closely related. 

I think one of the failures of the previous template was the number of sections and maybe we can compact that here?

File: meta/template.md
Comment (nkolev92) [inline meta/template.md]: Feels like the why and the problem section are very closely related, and can maybe be merged. 
I think we can provide `instructions` or guidelines as means of comments and not necessarily sections.

File: meta/template.md
Comment (nkolev92) [inline meta/template.md]: This one is...awkward :D
Do we talk about metrics?

File: meta/template.md
Comment (nkolev92) [inline meta/template.md]: Feels like this belong somewhere with the problem statement.

Comment (nkolev92) [reply meta/template.md]: Likewise, I think defining the scope of the changes is very important, hence why we used to have a goals/non-goals section. I'm not advocating for that, but ideally we have some sections that call out the non-goals in some way, shape or form.

File: meta/template.md
Comment (nkolev92) [reply meta/template.md]: We tend to continue iterating on designs that have already been PRed, so an open questions section allows the reader to follow what the design covers and what it does not. 

Now, that section is definitely no ideal. The goal of most designs will be to have an empty open questions section, so I'm not particularly sure how this can be captured.

File: meta/template.md
Comment (nkolev92) [inline meta/template.md]: The `What` and `How` feel like sections that could be combined.

**Review by nkolev92 — COMMENTED** [review]
I love that we keep iterating on this! This is making our repo more `presentable`. 

I do have some ideas about the template. 

I think our previous template suffered from having too many sections. Initial instinct is that maybe we're missing out on a few things. 

I do love the ehnanced descriptions of the sections in particular though.

File: meta/template.md
Comment (JonDouglas) [reply meta/template.md:48]: There's a great section here on what type of questions are needed:

http://smallcultfollowing.com/babysteps/blog/2018/06/20/proposal-for-a-staged-rfc-process/#the-proposal-point-and-the-change-from-spitballing-to-designing (Rust RFC process)

My approach was to use a typical 1-pager process for a proposal. If anything is more than 1 page as a proposal, it might be hard to understand.

I think the `Considerations` or `Cross Cutting Concerns` / etc should be a section?

For `References`, that's how I was treating it as well. It would be part of the other areas or in the design itself.

File: meta/template.md
Comment (JonDouglas) [reply meta/template.md]: They are definitely really closely related. 

Description -> Tell me a paragraph about it
Problem -> Tell me what problems it solves
Why -> Tell me why we should solve this problem

These three really help define the motivation & prioritization from the maintainer perspective in my opinion.

I think they can definitely be grouped, I just worry that we would imply people would know how to put that all under one section, so I broke it down like I usually do.

File: meta/template.md
Comment (JonDouglas) [reply meta/template.md]: How about a `Motivation` section?

File: meta/template.md
Comment (JonDouglas) [reply meta/template.md]: I think we get rid of it? I just think it helps people think about the "Why we're solving it" more deeply.

File: meta/template.md
Comment (nkolev92) [reply meta/template.md]: We can maybe consider Header 3 sections that way it's communicated that they are related. 

> I think they can definitely be grouped, I just worry that we would imply people would know how to put that all under one section, so I broke it down like I usually do.

I don't think there's harm in that :) I think it makes sense to break them in this particular way to an extent. I just see this and see a lot of `1 line` why sections, because the problem kind of covered that. 

My proposal would be, include the breakdown, but not by means of section headings, rather comments (either `<!-- here be comment -->` or just regular inline comment.

File: meta/template.md
Comment (nkolev92) [reply meta/template.md]: I really like `Motivation`!

File: meta/template.md
Comment (nkolev92) [reply meta/template.md]: I think this particular one is super relevant for our own product processes, especially ones that involve a lot of customer development, but isn't necessarily something that I expect would be easy to propose/follow for external contributors. 

So, I like the gist of `think about this`, but not sure we'd benefit from a section for it.

**Comment by JonDouglas** [comment]
Okay here's another proposal that is more rust inspired.

# Title

- Author Name
- Start Date
- GitHub Issue (If any)
- GitHub PR (If any)

## Summary

A concise, one-paragraph description of the proposal.

## Motivation 

- A short background of the problem and how your solution will solve it.
- A few details on why we're doing this, and what pain points this solves. Use real, concrete examples to make your case.
- A brief description to help everyone understand who this proposal is for.
- A couple ways on how we might measure the success or outcome of this proposal.

## Explanation

### Functional explanation

- A chance to introduce new concepts.
- A functional design explaining the proposal with examples.
- A low-fidelity mockup or pseudocode that shows how this proposal would look.

### Technical explanation

- A technical design explaining the proposal with implementation details.
- A clear interaction model with existing features.

## Drawbacks

- A consideration for not doing this.

## Alternatives / Cross Cutting Concerns

- A comparison of other possible designs.
- A considerations for other designs and the rationale of choosing them or not.
- An understanding of the impact of not doing this.

## Prior Art

- What features exist in other ecosystems and the experiences their communities have had with them?
- Has this been done by another community before?
- What lessons can we learn from other communities about it?
- Any great documentation, blogs, or papers that will help serve as a background?

## Unresolved Questions

- Bikeshedding

## Future Possibilities

- Does this unlock a future possibility or idea you have?

**Comment by nkolev92** [comment]
I personally like that one a lot better.

File: meta/template.md
Comment (nkolev92) [reply meta/template.md:48]: Sounds good to me.

**Comment by zkat** [comment]
I like the Rust one because it's already gone through a lot of iteration and it's been proven to be pretty successful ;)

File: meta/template.md
Comment (JonDouglas) [reply meta/template.md]: I'll include comments for the explanation sections in a new revision.

**Comment by JonDouglas** [comment]
Okay, template is updated. Please tear apart the language (especially in the comments to describe to the poster)

File: meta/template.md
Comment (nkolev92) [inline meta/template.md:18]: In the past we've never had a functional and technical separation, but I think it's worth a shot. 
Some NuGet features are almost *completely* technical. Others, not so much. 
This spec covers both.
Yay, trying out new things :)

File: meta/template.md
Comment (nkolev92) [inline meta/template.md]:
```suggestion
<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
```

**Review by nkolev92 — APPROVED** [review]
I like it!
**progress**!!

File: meta/template.md
Comment (JonDouglas) [reply meta/template.md:18]: Maybe we should make a note that the explanation itself doesn't have to be both?

Like you can have a proposal purely technical or you can have one purely functional?

**Review by loic-sharma — APPROVED** [review]

File: meta/template.md
Comment (nkolev92) [reply meta/template.md:18]: I don't mind that :)
Feel free to make a judgment call.

Comment (JonDouglas) [reply meta/template.md:18]: 🚢it is my motto

**Comment by JonDouglas** [comment]
2 approvals so far. Any further consideration before merge?

**Pull request merged**
